src/dsp_tools/utils/excel_to_json_resources.py

```diff
@@ -31,8 +32,26 @@ def _validate_resources_with_schema(resources_list: list[dict[str, Any]]) -> boo
     try:
         jsonschema.validate(instance=resources_list, schema=resources_schema)
     except jsonschema.ValidationError as err:
-        raise BaseError(f'"resources" section did not pass validation. The error message is: {err.message}\n'
-                        f'The error occurred at {err.json_path}') from None
+        err_msg = f"'resources' section did not pass validation. "
+        json_path_to_resource = regex.search(r"^\$\[(\d+)\]", err.json_path)
+        if json_path_to_resource:
+            wrong_resource_name = jsonpath_ng.ext.parse(json_path_to_resource.group(0)).find(resources_list)[0].value["name"]
+            affected_field = regex.search(r"name|labels|comments|super|cardinalities\[(\d+)\]", err.json_path)
+            if affected_field and affected_field.group(0) in ["name", "labels", "comments", "super"]:
+                excel_row = int(json_path_to_resource.group(1)) + 2
+                err_msg += f"The problem is that the Excel sheet 'classes' contains an invalid value for resource '{wrong_resource_name}', " \
+                           f"in row {excel_row}, column '{affected_field.group(0)}': {err.message}"
+            elif affected_field and "cardinalities" in affected_field.group(0):
+                excel_row = int(affected_field.group(1)) + 2
+                if err.json_path.endswith("cardinality"):
+                    err_msg += f"The problem is that the Excel sheet '{wrong_resource_name}' contains an invalid value " \
+                               f"in row {excel_row}, column 'Cardinality': {err.message}"
+                elif err.json_path.endswith("propname"):
+                    err_msg += f"The problem is that the Excel sheet '{wrong_resource_name}' contains an invalid value " \
+                               f"in row {excel_row}, column 'Property': {err.message}"
+        else:
+            err_msg += f"The error message is: {err.message}\nThe error occurred at {err.json_path}"
+        raise BaseError(err_msg) from None
```

Comment:
I see how this improves the user experience, and I'm ok with that, of course. But I think this again goes against the thing we once discusses, that metrics, logs, user feedback and error handling are/should be separate things. To me, this should clearly be a part of user feedback, not of error handling.

Reply:
I agree with what @BalduinLandolt says. The separation of metrics, logs and feedback would need a concept first. I think we still don't have it for DSP-TOOLS, right?

Reply:
No, we don't have such a concept.
A spontaneous thought: As discussed some time ago, I plan to distinguish between 
 - BaseError: internal, will (hopefully) be caught in the call hierarchy and handled. If not handled, it will be printed as stack trace
 - UserError: will interrupt the program flow immediately, and print the message without stack trace. 

This is the only sensible way how I can imagine a separation of errors and user feedback.

Reply:
IMO user feedback should not be an error at all - errors should be errors, user feedback should be displayed in whatever interface the user uses - so in the case of a CLI application, the command line. Hence if your program needs to know of an invalid state, it should represent it with an error, and if it wants to inform the user of something, it should print it to the console. (or maybe write it to a file, it that's the feedback channel)

Reply:
I partly understand the separation of the concepts, but I have some big question marks. Is it really possible to separate these two? In the present case, the program encounters an invalid state of the data. This is an error (?) So it must inform the user that his input data is invalid. So is this an error or user feedback?

Reply:
it should do both. The user feedback is mandatory and displayed instantly. And then the error state is created, which propagates outward in the call hierarchy, until it's either caught (if it's an error that is non fatal and can be recovered from) or it reaches the outmost layer and the program exits with a non-zero code.  
But that's the ideal scenario (or at least what I think it is), not something that needs doing right away